cmd/security-agent/app/app.go

```diff
@@ -259,7 +260,9 @@ func RunAgent(ctx context.Context, pidfilePath string) (err error) {
 		}
 	}
 
-	complianceAgent, err := compliance.StartCompliance(hostnameDetected, stopper, statsdClient)
+	var log complog.Component
+	var config compconfig.Component
+	complianceAgent, err := compliance.StartCompliance(log, config, hostnameDetected, stopper, statsdClient)
```

Comment:
Why do you pass uninitialise Log and Config components?

Reply:
That is a good catch. I originally broke this PR out from https://github.com/DataDog/datadog-agent/pull/14814 to make it easier to review, but I realized now this was break out was not done correctly. I'm going to merge this PR back into the main PR because it's going to be too complex to only go halfway with using the log and config components.